feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/PreviewScreen.kt

```diff
@@ -77,20 +70,6 @@ fun PreviewScreen(
     val screenFlashUiState: ScreenFlash.ScreenFlashUiState
         by viewModel.screenFlash.screenFlashUiState.collectAsState()
 
-    val deferredSurfaceProvider = remember { CompletableDeferred<SurfaceProvider>() }
-    var surfaceProvider by remember { mutableStateOf<SurfaceProvider?>(null) }
-
-    LaunchedEffect(LocalLifecycleOwner.current) {
-        surfaceProvider = deferredSurfaceProvider.await()
-    }
-
-    LifecycleStartEffect(surfaceProvider) {
-        surfaceProvider?.let { viewModel.runCamera(it) }
-        onStopOrDispose {
-            viewModel.stopCamera()
-        }
-    }
-
```

Comment:
Another option here that would remove recompositions would be:
```
    val surfaceProviderState = remember { MutableStateFlow<SurfaceProvider?>(null) }

    LifecycleStartEffect(Unit) {
        val startedJob = lifecycleScope.launch {
            surfaceProviderState
                .filterNotNull()
                .onCompletion { viewModel.stopCamera() }
                .collect { viewModel.runCamera(it) }
        }

        onStopOrDispose { startedJob.cancel() }
    }
```

and

```
onSurfaceProviderReady = { surfaceProviderState.value = it }
```

This would also allow the `ImplementationMode` to be changed, and would react correctly.

Of course, this logic can be simplified further if we move the SurfaceProvider state into the ViewModel.